File: src/test/java/org/yankov/mso/database/DatabaseConnectionTest.java
package org.yankov.mso.database;

import org.junit.Test;
import org.yankov.mso.application.ApplicationContext;

public class DatabaseConnectionTest extends DatabaseTest {

    @Test
    public void testDatabaseConnection() {
        ApplicationContext.getInstance().getDatabaseManager().executeOperation(null);
    }

}

File: src/test/scala/org/yankov/mso/application/database/DatabaseConnectionTest.scala
package org.yankov.mso.application.database

import org.scalatest.{BeforeAndAfterAll, Matchers, WordSpec}

class DatabaseConnectionTest extends WordSpec with Matchers with BeforeAndAfterAll {
 override def beforeAll(): Unit = System.setSecurityManager(null)

  "connect should succeed" in {
    val connectionString = ConnectionStringFactory.createDerbyConnectionString(
      InMemoryDatabaseProtocol,
      "test",
      Map("create" -> "true")
    )

    DatabaseConnection.connect(connectionString).isRight shouldBe true
  }
}
